### DISCUSSION TITLE
v1.1.0 Release Notes

### DISCUSSION THREAD

**Discussion by endigo9740** [skeletonlabs/skeleton#1215] [Announcements]
## News

The Skeleton v1.0 was a great success, thank you everyone for your kind words and joining us for this. This of course included two live stream events covering what Skeleton is, how to get started, and some advanced tips and tricks. If you missed it, check out the playlist, now with interactive chapter timestamps to jump straight to the topics you're interested in:

https://www.youtube.com/playlist?list=PLJtgPDPrJYAuDISpNrarLCjVInKknDCcA

## Release Notes

https://github.com/skeletonlabs/skeleton/releases/tag/1.1.0

As promised during our launch event, today's release contains primarily triage and bug fixes. However we've also implemented the new autocomplete component, made changes to preset theme fonts to ensure they are GDPR compliant, as well as some minor improvements to several features. See the full notes below.

- 💬 New autocomplete component
- ✅ Improved themes docs + GDPR compliance
- 🆕 Minor feature updates
- 🐞 Various bug fixes

## New Component: Autocomplete - https://github.com/skeletonlabs/skeleton/pull/1206

![Screenshot 2023-03-27 at 1 56 44 PM](https://user-images.githubusercontent.com/1509726/228039682-b3c4d116-c407-4dfc-af6f-39c58f0c5271.png)

Docs: https://www.skeleton.dev/components/autocomplete

We've finally introduced the long desired autocomplete component! We've taken our time because we wanted to make it as flexible and useful as possible. As such, this can directly interface with nearly any type of suggested input you might want, including direct support for the Input Chip component. It supports whitelists, blacklists, and even keywords for fuzzy search. Plus it pairs directly with the Popup utility. We think you're going to love it!

## Themes Doc + GDPR Compliance - https://github.com/skeletonlabs/skeleton/issues/1161

![Screenshot 2023-03-27 at 1 57 31 PM](https://user-images.githubusercontent.com/1509726/228039908-1f99d735-6350-4c8c-853e-eba1a7d5542e.png)

View the new Theme docs page: https://www.skeleton.dev/docs/themes

If you're using preset themes you'll note that custom fonts will initially be lost when upgrading to v1.1.0. It was brought to our attention that embedding Google Fonts within our themes poses a risk to GDPR compliance. Given this, we've made a few notable changes:

1. Fonts are no longer imported directly within each preset Skeleton theme. You must import them in your project.
2. For GDPR compliance (and better performance) we recommend locally installing and importing fonts.
3. Finally, we've introduced a new [Docs > Themes](https://www.skeleton.dev/docs/themes) section which covers a number of ways to customize your themes, including detailed instruction for local or remote imports of custom fonts!

Additionally the new Themes page can be tailored to your specific theme and all instruction will be update accordingly. This details each font to download, then how and where to install and import them.

> **Warning** Please be aware we have a pending update to the CLI that will handle the fonts on install. In the meantime new CLI projects will be missing their custom fonts. You can, however, [handle this manually following these instructions](https://www.skeleton.dev/docs/themes#custom-fonts).

## Popup Focus Event - https://github.com/skeletonlabs/skeleton/pull/1174

![Screenshot 2023-03-27 at 2 01 12 PM](https://user-images.githubusercontent.com/1509726/228040662-efd19874-6a62-4276-bbb2-4e87b438503a.png)

The popup utility now supports two new event types: `focus` and `focus-click`. The former allows a popup to appear when you focus an element, such as an input, then close when you click outside this element. While the latter works the same, except clicking the trigger element repeatedly will toggle the popup. This is the recommended event type to use alongside the new Autocomplete component as it keeps the focus on the input for search, rather than jumping into the focusable elements within the popup.

## Minor Improvements

- The Avatar component now supports `$$restProps` for implementing arbitrary image tag attributes:
https://github.com/skeletonlabs/skeleton/issues/1185

### Bugfixes

- Fixed an issue prevent File Dropzone binding:
https://github.com/skeletonlabs/skeleton/issues/1021
- The complete button in the Stepper now handles the locked state:
https://github.com/skeletonlabs/skeleton/issues/1164
- Removed a redundant space class from the App Bar default styles:
https://github.com/skeletonlabs/skeleton/issues/1171
- Resolved a bug that prevents the Popup action from reflecting live updates:
https://github.com/skeletonlabs/skeleton/issues/1186
- Resolved a bug where the Prompt modal was not capturing form submission properly:
https://github.com/skeletonlabs/skeleton/issues/1193
- Fixed a bug where the AppRailTile `on:click` event would trigger twice:
https://github.com/skeletonlabs/skeleton/issues/1195

### Documentation

- Adjusted permalinks and table of contents:
https://github.com/skeletonlabs/skeleton/pull/1144
- Documented how to pin to manually pin to dark mode:
https://github.com/skeletonlabs/skeleton/issues/1146
- Fixed an animation issue with the new documentation preview component:
https://github.com/skeletonlabs/skeleton/issues/1148
- Updated Prettier config to aid contributors:
https://github.com/skeletonlabs/skeleton/pull/1162
- A direct link to the Skeleton YouTube channel is now in the page footer:
https://github.com/skeletonlabs/skeleton/pull/1169
- Fix several code snippets on the Toast documentation:
https://github.com/skeletonlabs/skeleton/issues/1172
- Documented the use case for z-index within the Popup utility docs:
https://github.com/skeletonlabs/skeleton/pull/1182
- Improved documentation per the Stepper's `stepTerm` property:
https://github.com/skeletonlabs/skeleton/pull/1203
- Resolved a number of typos:
https://github.com/skeletonlabs/skeleton/issues/1134
https://github.com/skeletonlabs/skeleton/issues/1140
https://github.com/skeletonlabs/skeleton/pull/1155
https://github.com/skeletonlabs/skeleton/pull/1198
https://github.com/skeletonlabs/skeleton/pull/1199
https://github.com/skeletonlabs/skeleton/pull/1208
https://github.com/skeletonlabs/skeleton/issues/1207

**Comment by saostad** [comment]
Great Job! 
I love everything Skeleton has.
